internal/gitutil/git.go

```diff
@@ -69,7 +70,27 @@ func EnsureUpdated(uri, destinationPath string) error {
 	return updateAndCleanUntracked(destinationPath)
 }
 
-func exec(pwd string, args ...string) error {
+// ListModifiedFiles will fetch origin and list modified files
+func ListModifiedFiles(uri, destinationPath string) ([]string, error) {
```

Comment:
`ListModifiedFiles` is a slight misnomer, because no files are actually modified. What about `ListUpstreamChanges`?